### DISCUSSION TITLE
Maya USD Import Help

### DISCUSSION THREAD

**Discussion by jcoelho88** [Autodesk/maya-usd#2214] [General] [answered]
Hello,

I want to be able to import my USD assets with variant selection via python.

When importing with the import dialog there is a option to open a hierarchy view and select the variants on what ever prim you want.

The command pops up in the script editor when its ran but I do not see how any of the options chosen in the Hierarchy view is being applied.

Currently an import command I use could look like this.

```python
cmds.file(
            assetusd_path, i=True, type='USD Import', ignoreVersion=True,
            renameAll=True, mergeNamespacesOnClash=False, namespace=namespace,
            options=";shadingMode=[[none,default]];preferredMaterial=none;primPath=/;readAnimData=0;"
                    "useCustomFrameRange=0;startTime=0;endTime=0;importUSDZTextures=0",
            preserveReferences=True, importFrameRate=True, importTimeRange='override',
        )
```

How do I include the options for variant selection on specific prims via the python import command?

Any help would be appreciated.

Cheers,

Jason Coelho

**Comment by JGamache-autodesk** [comment, marked as answer]
Hi @jcoelho88, all the information is stored in a singleton class called [ImportData](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/fileio/importData.h).

That class is not yet exported to the Python API. Would you like to wrap it and submit a PR? See files in [lib/mayaUsd/python](https://github.com/Autodesk/maya-usd/tree/dev/lib/mayaUsd/python) for Python wrapper examples. See wrapStageCache.cpp for a starting point on exporting singletons.

**Reply by jcoelho88** [reply]
Hi @JGamache-autodesk 

This would be new for me but I'm up for it.

I'll take a look at the references and the time commitment, if its something I can do I will do so and submit a PR.

Thanks for the info.

Cheers,

Jason Coelho

**Reply by erikstantoncarlson** [reply]
Hi @JGamache-autodesk . Has there been any updates on this?
It currently seems impossible to choose the variants that get imported when scripting with Python.

**Reply by neilh-adsk** [reply]
Nothing further has been added on this front.  Is this something you would still consider trying to help us expose ?  I'll make sure it gets added to the backlog or prioritised if it is already there.  @tanant, something to followup on.

Neil

**Reply by erikstantoncarlson** [reply]
I wish I could expose it myself, but I don't have the technical skills to compile the plugin.

**Reply by tanant** [reply]
@erikstantoncarlson @jcoelho88 minor admin - i've just converted this back to an issue (well, made an issue from this discussion) so it's more visible and can be prio'd etc, I think it's better in this case for us to do it as an issue - #2685 - if you want to comment there with any details on what you'd possibly expect that interface to look like or detail i've missed here go for it!

**Reply by jcoelho88** [reply]
Hello @tanant,

Thanks for making it an issue, I did attempt what @JGamache-autodesk but it was beyond what I was capable of doing. I had issues even building the plugin!

I will be sure to take a look at the new issue.

Cheers,

Jason Coelho
